src/components/Modal/Modal.tsx

```diff
@@ -52,8 +52,8 @@ export function Modal(props: ModalProps) {
       onClose: closeModal,
       isDismissable: true,
       shouldCloseOnInteractOutside: (el) => {
-        // Do not close the Modal if the user is interacting with the Tribute mentions dropdown (via RichTextField).
-        return !el.closest(".tribute-container");
+        // Do not close the Modal if the user is interacting with the Tribute mentions dropdown (via RichTextField) or with another dialog on top of it.
```

Comment:
Maybe "another 3rd party dialog like lightbox" to differentiate its not a beam dialog.